tests/MongoDB.Driver.Tests/FilterDefinitionBuilderTests.cs

```diff
@@ -78,6 +78,32 @@ public void And_with_clashing_keys_but_different_operators_should_get_merged()
             Assert(filter, "{a: {$gt: 1, $lt: 10}}");
         }
 
+        [Theory]
+        [InlineData("{ geoField : { $geoWithin : { $box : [ [ 1.0, 2.0 ], [ 3.0, 4.0 ] ] } } }", "{ geoField : { $near : [ 5.0, 6.0 ] } }")]
+        [InlineData("{ geoField : { $near : [ 5.0, 6.0 ] } }", "{ geoField : { $geoWithin : { $box : [ [ 1.0, 2.0 ], [ 3.0, 4.0 ] ] } } }")]
+        [InlineData("{ geoField : { $nearSphere : { $geometry : { type : 'Point', coordinates : [ 1, 2 ] } } } }", "{ geoField : { $geoIntersects : { $geometry : { type : 'Polygon', coordinates: [ [ [ 1, 2 ], [ 3, 4 ], [ 5, 6 ], [ 7, 8 ] ] ] } } } }")]
+        [InlineData("{ geoField : { $geoIntersects : { $geometry : { type : 'Polygon', coordinates: [ [ [ 1, 2 ], [ 3, 4 ], [ 5, 6 ], [ 7, 8 ] ] ] } } } }", "{ geoField : { $nearSphere : { $geometry : { type : 'Point', coordinates : [ 1, 2 ] } } } }")]
+        public void And_with_clashing_keys_and_different_operators_but_with_filters_that_support_only_dollar_form_should_get_promoted_to_dollar_form(string firstFilter, string secondFilter)
+        {
+            var subject = CreateSubject<BsonDocument>();
+
+            var combinedFilter = subject.And(firstFilter, secondFilter);
+
+            Assert(combinedFilter, $"{{ $and : [ {firstFilter}, {secondFilter} ] }}");
```

Comment:
Nit: Should we use fluent assertions here?

Reply:
in general yes, but this is the custom method which is defined in this class. So, in this particular case - no :)

Reply:
Whoops! Thanks for the explanation! :)